### ISSUE TITLE
Can I request 2 functions?

### ISSUE THREAD

**Issue by MheadHero** [sovit-123/fastercnn-pytorch-training-pipeline#15]
Function 1 is to output validation loss plot using tensorboard
Function 2 is make something like results.csv to shows all the mAP on every epoch, so that we can know at which epoch we get the best mAP. 

Example for result.csv is as figure below. This results.csv is from yolov5 one.

![image](https://user-images.githubusercontent.com/24875124/174480875-62f6d3b7-2fce-4a27-ac18-caf6a73ab52f.png)

**Comment by sovit-123** [comment]
I will surely try to add the second feature.
Regarding the first feature, I am not very sure. The reason is that all of the Faster RCNN models here follow the PyTorch model builder pipeline which does not output a validation loss by default. It's not just this repository but also many others like MMDetection and YOLO Ultralytic models. Also, changing the code to do so, is going out of the current pipeline a bit which may break a few things. **But I can surely guarantee adding the first feature.**

**Comment by MheadHero** [comment]
That's mean you will add both the functions? Many thanks sovit!

**Comment by sovit-123** [comment]
Hi. I can guarantee that I can add the second feature for saving the mAP in a file in some optimized format. But I am a bit skeptical about the first feature. The thing is that I don't want to change the internal working of the official PyTorch model.

**Comment by MheadHero** [comment]
What about this? It seems like they have solved the problem. 
https://stackoverflow.com/questions/71288513/how-can-i-determine-validation-loss-for-faster-rcnn-pytorch

**Comment by sovit-123** [comment]
So that would require breaking the model into different components like the features (classifier) module, RPN module, and the detection module. I am unsure whether this should be done and how it will affect future updates to the code base. I can try this but mostly I will take this on a low-priority basis. The main reason is that in object detection, in almost all cases, the evaluation mAP should be used to choose and save the best model. I may try this but I cannot guarantee a timeline for this now. I hope you understand the situation.

**Comment by sovit-123** [comment]
Also, it may be some time before I start testing this out. But if you wish, you can create a PR with the code (you can take your time developing it) and if everything works out fine, I will be quite happy to merge it.

**Comment by sovit-123** [comment]
@MheadHero 
The code has been updated. Now a `results.csv` gets saved which contains the mAP @0.5:0.95 IoU and mAP @0.5 IoU.

**Comment by MheadHero** [comment]
@sovit-123 yaya, I saw that function. Thank you again. 


> @MheadHero Also, it may be some time before I start testing this out. But if you wish, you can create a PR with the code (you can take your time developing it) and if everything works out fine, I will be quite happy to merge it.

Alrighttt, I will try and develop it.

**Comment by kalikhademi** [comment]
Could you please point me to the name of this function that will compute the results.csv?

**Comment by sovit-123** [comment]
@kalikhademi 
They are `csv_log` and `create_log_csv` in logging.py
https://github.com/sovit-123/fasterrcnn-pytorch-training-pipeline/blob/main/utils/logging.py
